catalog/s3-policies.yaml

```diff
@@ -16,29 +15,68 @@
   resources:
     - "*"
 
-
 - sid: "DenyS3IncorrectEncryptionHeader"
   effect: "Deny"
   actions:
     - "s3:PutObject"
-  resources:
-    - "*"
   condition:
     - test: "StringNotEquals"
       variable: "s3:x-amz-server-side-encryption"
       values:
         - "AES256"
         - "aws:kms"
-
+  resources:
+    - "*"
 
 - sid: "DenyS3UnEncryptedObjectUploads"
   effect: "Deny"
   actions:
     - "s3:PutObject"
-  resources:
-    - "*"
   condition:
     - test: "Null"
       variable: "s3:x-amz-server-side-encryption"
       values:
         - true
+  resources:
+    - "*"
+
+- sid: "DenyNoTLSRequests"
+  effect: "Deny"
+  actions:
+    - "s3:*"
+  condition:
+    - test: "Bool"
+      variable: "aws:SecureTransport"
+      values:
+        - false
+  resources:
+    - "arn:aws:s3:::*/*"
+
+- sid: "RequireMFAS3Delete"
+  effect: "Deny"
+  actions:
+    - "s3:DeleteObject"
+  condition:
+    - test: "BoolIfExists"
+      variable: "aws:MultiFactorAuthPresent"
+      values:
+        - false
+  resources:
+    - "arn:aws:s3:::*/*"
```

Comment:
We should be careful when applying this policy since our customers are generally using federated authentication, which won't have the MFA tag on the session. With this policy in place, the only way to delete an S3 bucket would be to use an IAM user with MFA enabled.

Reply:
good point, I think if this is going to be the case of most of our customers then we can delete this specific SCP